services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentContentService.kt

```diff
@@ -19,18 +19,26 @@ class ApplicationAttachmentContentService(
     val contentRepository: ApplicationAttachmentContentRepository,
     val fileClient: FileClient,
 ) {
-    fun save(attachmentId: UUID, content: ByteArray) {
-        contentRepository.save(ApplicationAttachmentContentEntity(attachmentId, content))
+    fun save(
+        filename: String,
+        contentType: MediaType,
+        applicationId: Long,
+        content: ByteArray
+    ): String {
+        val blobPath = generateBlobPath(applicationId)
+        fileClient.upload(Container.HAKEMUS_LIITTEET, blobPath, filename, contentType, content)
+        return blobPath
     }
 
-    fun delete(attachment: AttachmentMetadata) =
-        with(attachment) {
-            logger.info { "Deleting attachment $id if blob defined. Blob=$blobLocation." }
-            blobLocation?.let { fileClient.delete(Container.HAKEMUS_LIITTEET, it) }
+    fun delete(blobPath: String?) {
+        if (blobPath == null) {
+            logger.info { "Blob path is null, nothing to delete" }
+            return
         }
+        fileClient.delete(Container.HAKEMUS_LIITTEET, blobPath)
+    }
```

Comment:
I think using `?.let` and the Elvis operator gives us cleaner code here.
```suggestion
    fun delete(blobPath: String?) {
        blobPath?.let { fileClient.delete(Container.HAKEMUS_LIITTEET, it) }
            ?: logger.info { "Blob path is null, nothing to delete" }
    }
```

If the log row isn't mandatory, this could be almost a oneliner:
```
    fun delete(blobPath: String?) {
        blobPath?.let { fileClient.delete(Container.HAKEMUS_LIITTEET, blobPath) }
    }
```

Then again, this is called from 3 places, and the blob location can be null in only one of those. So the null case could be also checked at the callsite.

Reply:
Made blobPath non-nullable and added a check in that one place where it is needed (`ApplicationAttachmentService.deleteAttachmend(attachmentId: UUID)`):
```
        attachment.blobLocation?.let { attachmentContentService.delete(it) }
            ?: logger.warn { "Attachment ${attachment.id} has no blob content" }
```
Not sure whether it should log at WARN level or INFO.

Reply:
Ok, changed it to INFO